codesigndoc/codesigngroup.go

```diff
@@ -68,83 +68,138 @@ func printCodesignGroup(group export.IosCodeSignGroup) {
-	}
+		log.Debugf("selected export method: %v", selectedExportMethod)
 
-	selectedCertificate := certificateutil.CertificateInfoModel{}
+		// Export method needs Development or Distribution certificate?
+		exportDistCert := true
+		if selectedExportMethod == "development" {
+			exportDistCert = false
+		}
 
-	if isDistributionCertificate(archiveCertificate) {
-		certificates := certificatesByTeam[selectedTeam]
-		developmentCertificates := certificateutil.FilterCertificateInfoModelsByFilterFunc(certificates, func(certInfo certificateutil.CertificateInfoModel) bool {
-			return !isDistributionCertificate(certInfo)
-		})
+		selectedTeam := ""
+		log.Debugf("InstalledCerts: %v\n", installedCertificates)
 
-		certificateOptions := []string{}
-		for _, certInfo := range developmentCertificates {
-			certificateOption := fmt.Sprintf("%s [%s]", certInfo.CommonName, certInfo.Serial)
-			certificateOptions = append(certificateOptions, certificateOption)
+		// Filter the installed certificates by distribution type
+		var certsForSelectedExport []certificateutil.CertificateInfoModel
+		if exportDistCert {
+			certsForSelectedExport = certificateutil.FilterCertificateInfoModelsByFilterFunc(installedCertificates, func(certInfo certificateutil.CertificateInfoModel) bool {
+				return isDistributionCertificate(certInfo)
+			})
+			log.Debugf("Distribution certificated: %v\n", certsForSelectedExport)
+		} else {
+			certsForSelectedExport = certificateutil.FilterCertificateInfoModelsByFilterFunc(installedCertificates, func(certInfo certificateutil.CertificateInfoModel) bool {
+				return !isDistributionCertificate(certInfo)
+			})
+			log.Debugf("Developer certificated: %v\n", certsForSelectedExport)
 		}
 
-		fmt.Println()
-		question := fmt.Sprintf(`The Xcode archive used distribution certificate: %s [%s].
-Please select a development certificate:`, archiveCertificate.CommonName, archiveCertificate.Serial)
-		selectedCertificateOption, err := goinp.SelectFromStringsWithDefault(question, 1, certificateOptions)
-		if err != nil {
-			return certificateutil.CertificateInfoModel{}, fmt.Errorf("failed to read input: %s", err)
+		certificatesByTeam := mapCertificatesByTeam(certsForSelectedExport)
+		log.Debugf("Filtered certificates (by distribution type) by team: %v\n", certificatesByTeam)
+
+		if len(certificatesByTeam) == 0 {
+			log.Warnf("🚨 We couldn't find any certificate for the %s export method.", selectedExportMethod)
+			continue
 		}
 
-		for _, certInfo := range developmentCertificates {
-			certificateOption := fmt.Sprintf("%s [%s]", certInfo.CommonName, certInfo.Serial)
-			if certificateOption == selectedCertificateOption {
-				selectedCertificate = certInfo
+		useArchiveTeam := true
+		_, contains := certificatesByTeam[fmt.Sprintf("%s - %s", archiveCertificate.TeamID, archiveCertificate.TeamName)]
+
+		// Skip if only 1 team has certificates on the machine.
+		if len(certificatesByTeam) > 1 && contains {
```

Comment:
warning if the archiving team does not have a certificate for the selected method.